### ISSUE TITLE
openresty blogging platform

### ISSUE THREAD

**Issue by despiegk** [threefoldtecharchive/jumpscaleX_archived#171]
- choose nice template for blogging (check with kristof/roel)
- make openresty plugin to get content of the blogs from gedis backend in JSX
- feed the content in the blog template
- support tags, ... (think about navigation), metadata comes from gedis
structure inside (wordcloud, categories, ...)
- try on blogs from ThreeFold Foundation
- support search (search implemented in the JSX backend in gedis)

how

- create/use gedis prototype and ask kds to review
- lua openresty or lapis backend
- use nice template
- use redis client in lua

**Comment by abom** [comment]
> make openresty plugin to get content of the blogs from gedis backend in JSX

Any preferred storage engine?

> try on blogs from ThreeFold Foundation

Need more input here, a blogging platform is not a static site, maybe you mean importing blogs from info_foundation repository?

> support search (search implemented in the JSX backend in gedis)

We have sonic backend now, should we use it?

**Comment by abom** [comment]
The initial idea is to make use of our tools (e.g. open publish) and host the posts on a git repository, other metadata like authors, modification dates...etc can be grabbed from the repository too.

**Comment by abom** [comment]
In case we are using a git repository, we need to force specific directory structure (maybe some metadata files too), to allow archiving, navigation...etc, also IMHO, other operations like indexing, auto-pulling...etc and user management won't be the best fit for blogging use case.

**Comment by abom** [comment]
I asked @VonSub for templates.

**Comment by despiegk** [comment]
index we should do in sonix search/index engine, there is a client for it in JSX
we already use some way how to embed toml data inside the .md files for some metadata like tags

**Comment by despiegk** [comment]
but no need to worry too much on the storage side of things, thats super easy to implement using our docgenerator, 
see the specs it says the information comes from Gedis backend which is an actor at the end, you can create a nice interface where the content comes from and I don't mind to review the POC implementation

**Comment by abom** [comment]
For the prototype, the following are in-progress
- [ ] lapis app with etlua templates for the blog (using this [free blog template](https://github.com/BlackrockDigital/startbootstrap-clean-blog) until @VonSub or @ehab-hassan give their word).

- [ ] gedis actor to load posts (will be md files from a repository) and metadata, which include some information that will be stored in git history, like:
  - Authors
  - Modification dates

  and other information that user provides via data sections (toml) like:
    - categories
    - tags

- [ ] search using sonic (should use our sonic client directly or the one integrated in in markdowndocs).

**Comment by abom** [comment]
@VonSub Proposed this template https://colorlib.com/wp/template/wordsmith/

**Comment by rkhamis** [comment]
Let's use our existing tools for wiki publishing to create blog-like behavior.

We can add in the blog's metadata information about its **date** of creation. This metadata will be provided by the content creator or git history (which will be a lot less efficient, but can be done one time only to populate the metadata). 
I would rather the content creator sets this data as part of their publishing to have full clarity of how the blog will be sorted.

We'll add a few features on top of wikis:
- archives macro: will provide easily navigatable blog entries by date
- tags: also from blog metadata. Will provide a tag cloud that will find relevant blog posts
- link to previous and next blog post by date


@abom want to add anything?

**Comment by abom** [comment]
Had some problems with schema loading (#491) and cloning repositories from github (still investigating).

The things i'm trying to do are:

* archive macro (to generate archive links, which are links to a collection of blog posts aggregated by date), also to generate tags.

Links and tags can be generated as a document with links or as a custom [navbar](https://docsify.now.sh/custom-navbar) for docsify.

**Comment by abom** [comment]
Blocked by #507.

**Comment by despiegk** [comment]
see new story about blog, jimber guys are working on it
